### PR TITLE
Add support for pruning analysis plugins output files

### PR DESCRIPTION
**Pull request by bbinet**

### CODE REVIEW

File: sandboxes/input/prune_input.lua [hunk 1]
```diff
@@ -53,19 +63,21 @@ function process_message()
...
-                end
-            else
-                off = t.off
-                if min ~= t.min then
-                    min = t.min
-                    for i = min - 1, 0, -1 do
-                        local r = os.remove(string.format("%s/input/%d.log", output_path, i))
-                        if not r then break end
+        for name, grammar in pairs(grammars) do
+            local t = grammar:match(s)
```
Comment (trink) [inline sandboxes/input/prune_input.lua]: I don't like parsing the file twice

File: sandboxes/input/prune_input.lua [hunk 2]
```diff
@@ -53,19 +63,21 @@ function process_message()
...
+            if t then
+                if min == t.min and off == t.off then
+                    if exit_on_stall then
```
Comment (trink) [inline sandboxes/input/prune_input.lua]: This is now broken, it should only exit when all data has stalled.

File: sandboxes/input/prune_input.lua (hunk 1 above)
Comment (bbinet) [inline sandboxes/input/prune_input.lua]: I see, I'll try to fix that

File: sandboxes/input/prune_input.lua (hunk 2 above)
Comment (bbinet) [inline sandboxes/input/prune_input.lua]: You're right, I'll fix that too.

**Comment by bbinet** [comment]
@trink I've updated this PR and addressed your comments

File: sandboxes/input/prune_input.lua [hunk 3]
```diff
@@ -30,21 +33,35 @@ l.locale(l)
...
+end
 
-local function get_min(t, i, o)
-    if not t.min then t.min = math.huge end
-    if i < t.min then
-        t.min = i
-        t.off = o
-     end
+local function get_min(t, n, i, o)
+    local elt = t[n] or {}
```
Comment (trink) [inline sandboxes/input/prune_input.lua]: could just initialize min in the table constructor

**Comment by trink** [comment]
Everything else looks good

File: sandboxes/input/prune_input.lua (hunk 3 above)
Comment (bbinet) [inline sandboxes/input/prune_input.lua]: right, this is now fixed in bb20b23

**Comment by trink** [comment]
Thanks

**Pull request merged**